src/model/trip-points-model.js

```diff
@@ -1,34 +1,79 @@
-import { mockPoints } from '../mock/mock-points';
+
 import Observable from '../framework/observable';
+import { UpdateType } from '../const';
 
 export default class TripPointsModel extends Observable {
-  constructor() {
+  #pointsApiService = null;
+  #points = new Map();
+
+  #isReady = false;
+
+  constructor({
+    pointsApiService
+  }) {
     super();
-    mockPoints.forEach((point)=>{
-      this.#points.set(
-        point.id, {
-          'base_price': point.base_price,
-          'date_from':  point.date_from,
-          'date_to': point.date_to,
-          'destination': point.destination,
-          'is_favorite': point.is_favorite,
-          'offers': point.offers,
-          'type': point.type
-        });
 
-    });
+
+    this.#pointsApiService = pointsApiService;
+
+
   }
 
-  #points = new Map();
+  async init () {
+    try {
+      const points = await this.#pointsApiService.points;
+      points.forEach((point)=>this.#points.set(this.#adaptPointToClient(point)[0],this.#adaptPointToClient(point)[1]));
+      this.#isReady = true;
+    } catch(err){
+      throw new Error('Can\'t get points');
+    }
+    this._notify('', UpdateType.INIT);
+  }
+
+  isPointsReady() {
+    return this.#isReady;
+  }
+
+
+  #adaptPointToServer (point) {
+    return {
+      'id':point[0],
+      'base_price':Number(point[1].basePrice),
+      'date_from':point[1].dateFrom,
+      'date_to':point[1].dateTo,
+      'is_favorite':point[1].isFavorite,
+      'offers': point[1].offers,
+      'type': point[1].type,
+      'destination': point[1].destination,
+
+    };
+  }
```

Comment:
```suggestion
  #adaptPointToServer ([id, point]) {
    return {
      'id': id,
      'base_price': Number(point.basePrice),
      'date_from': point.dateFrom,
      'date_to': point.dateTo,
      'is_favorite': point.isFavorite,
      'offers': point.offers,
      'type': point.type,
      'destination': point.destination,
    };
  }
```
Можно использовать деструктуризацию для более удобного именования.